daemon/emer-persistent-cache.c

```diff
@@ -1051,29 +1051,47 @@ emer_persistent_cache_read (EmerPersistentCache *self,
     emer_persistent_cache_get_instance_private (self);
 
   GBytes **elems;
-  gsize num_elems;
+  GVariant **local_variants;
   guint64 local_token;
-  gboolean read_succeeded =
+  gboolean corrupt_data;
+  gboolean read_succeeded;
+  gsize num_elems;
+  gsize i;
+
+  read_succeeded =
     emer_circular_file_read (priv->variant_file, &elems, cost, &num_elems,
                              &local_token, error);
   if (!read_succeeded)
     return FALSE;
 
-  GVariant **local_variants = g_new (GVariant *, num_elems);
-  for (gsize i = 0; i < num_elems; i++)
+  corrupt_data = FALSE;
+  local_variants = g_new (GVariant *, num_elems);
```

Comment:
Oh! My mistake. I was under the impression that it was using the C89 way of declaring everything at the beginning of the block. will change it